src/message.ts

```diff
@@ -466,22 +467,19 @@ export class Message implements Sayable {
             name = mentionName + '@' + name
             nameList.push(name.slice(0, -1)) // get rid of the `@` at beginning
           })
-        return nameList
+        mentionList = mentionList.concat(nameList || [])
+        return mentionList
     }
 
-    // flatten array, see http://stackoverflow.com/a/10865042/1123955
-    const mentionList = [].concat.apply([], rawMentionedList)
     log.verbose('Message', 'mentioned(%s),get mentionList: %s', this.content(), JSON.stringify(mentionList))
 
-    contactList = [].concat.apply([],
-      mentionList.map(nameStr => {
-        room.memberAll(nameStr)
-      })
-      .filter(contact => !!contact),
-    )
+    mentionList.map(nameStr => {
+      const nameStrList = room.memberAll(nameStr)
```

Comment:
variable name `nameStrList` is not right: it's a type of `Contact`, not `String`.